### ISSUE TITLE
yarn install error

### ISSUE THREAD

**Issue by DrSkyfaR** [mastodon/mastodon#18607]
### Steps to reproduce the problem

1. yarn install for upgrade to v3.5.3

### Expected behaviour

Installing the dependencies without error message

### Actual behaviour

However, installs the dependencies with the error message incompatible Module

### Specifications

Hello all, 
I received the following error messages with the upgrade:

> yarn install
> yarn install v1.22.19
> [1/6] Validating package.json...
> [2/6] Resolving packages...
> [3/6] Fetching packages...
> error eslint-plugin-promise@6.0.0: The engine "node" is incompatible with this module. Expected version "^12.22.0 || ^14.17.0 || >=16.0.0". Got "12.13.0"
> error Found incompatible module.
> info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.

> RAILS_ENV=production bundle exec rails assets:precompile
> yarn install v1.22.19
> [1/6] Validating package.json...
> [2/6] Resolving packages...
> [3/6] Fetching packages...
> error eslint-plugin-promise@6.0.0: The engine "node" is incompatible with this module. Expected version "^12.22.0 || ^14.17.0 || >=16.0.0". Got "12.13.0"
> error Found incompatible module.
> info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
> Compiling...
> Compilation failed:
> ModuleNotFoundError: Module not found: Error: Can't resolve 'fuzzysort' in '/home/mastodon/live/app/javascript/mastodon/features/compose/components'
>     at /home/mastodon/live/node_modules/webpack/lib/Compilation.js:925:10
>     at /home/mastodon/live/node_modules/webpack/lib/NormalModuleFactory.js:401:22
>     at /home/mastodon/live/node_modules/webpack/lib/NormalModuleFactory.js:130:21
>     at /home/mastodon/live/node_modules/webpack/lib/NormalModuleFactory.js:224:22
>     at /home/mastodon/live/node_modules/neo-async/async.js:2830:7
>     at /home/mastodon/live/node_modules/neo-async/async.js:6877:13
>     at /home/mastodon/live/node_modules/webpack/lib/NormalModuleFactory.js:214:25
>     at /home/mastodon/live/node_modules/enhanced-resolve/lib/Resolver.js:213:14
>     at /home/mastodon/live/node_modules/enhanced-resolve/lib/Resolver.js:285:5
>     at eval (eval at create (/home/mastodon/live/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
>     at /home/mastodon/live/node_modules/enhanced-resolve/lib/UnsafeCachePlugin.js:44:7
>     at /home/mastodon/live/node_modules/enhanced-resolve/lib/Resolver.js:285:5
>     at eval (eval at create (/home/mastodon/live/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
>     at /home/mastodon/live/node_modules/enhanced-resolve/lib/Resolver.js:285:5
>     at eval (eval at create (/home/mastodon/live/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:27:1)
>     at /home/mastodon/live/node_modules/enhanced-resolve/lib/DescriptionFilePlugin.js:67:43
>     at /home/mastodon/live/node_modules/enhanced-resolve/lib/Resolver.js:285:5
>     at eval (eval at create (/home/mastodon/live/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:43:1)
>     at /home/mastodon/live/node_modules/enhanced-resolve/lib/ModuleKindPlugin.js:30:40
>     at /home/mastodon/live/node_modules/enhanced-resolve/lib/Resolver.js:285:5
>     at eval (eval at create (/home/mastodon/live/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
>     at /home/mastodon/live/node_modules/enhanced-resolve/lib/Resolver.js:285:5
>     at eval (eval at create (/home/mastodon/live/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:16:1)
>     at /home/mastodon/live/node_modules/enhanced-resolve/lib/forEachBail.js:30:14
>     at /home/mastodon/live/node_modules/enhanced-resolve/lib/Resolver.js:285:5
>     at eval (eval at create (/home/mastodon/live/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
>     at /home/mastodon/live/node_modules/enhanced-resolve/lib/UnsafeCachePlugin.js:44:7
>     at /home/mastodon/live/node_modules/enhanced-resolve/lib/Resolver.js:285:5
>     at eval (eval at create (/home/mastodon/live/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
>     at /home/mastodon/live/node_modules/enhanced-resolve/lib/Resolver.js:285:5
>     at eval (eval at create (/home/mastodon/live/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:27:1)
>     at /home/mastodon/live/node_modules/enhanced-resolve/lib/DescriptionFilePlugin.js:67:43
>     at /home/mastodon/live/node_modules/enhanced-resolve/lib/Resolver.js:285:5
>     at eval (eval at create (/home/mastodon/live/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:16:1)
>     at /home/mastodon/live/node_modules/enhanced-resolve/lib/RootPlugin.js:37:38
>     at _next42 (eval at create (/home/mastodon/live/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:8:1)
> resolve 'fuzzysort' in '/home/mastodon/live/app/javascript/mastodon/features/compose/components'
>   Parsed request is a module
>   using description file: /home/mastodon/live/package.json (relative path: ./app/javascript/mastodon/features/compose/components)
>     Field 'browser' doesn't contain a valid alias configuration
>     resolve as module
>       looking for modules in /home/mastodon/live/app/javascript
>         using description file: /home/mastodon/live/package.json (relative path: ./app/javascript)
>           Field 'browser' doesn't contain a valid alias configuration
>           using description file: /home/mastodon/live/package.json (relative path: ./app/javascript/fuzzysort)
>             no extension
>               Field 'browser' doesn't contain a valid alias configuration
>               /home/mastodon/live/app/javascript/fuzzysort doesn't exist
>             .mjs
>               Field 'browser' doesn't contain a valid alias configuration
>               /home/mastodon/live/app/javascript/fuzzysort.mjs doesn't exist
>             .js
>               Field 'browser' doesn't contain a valid alias configuration
>               /home/mastodon/live/app/javascript/fuzzysort.js doesn't exist
>             .sass
>               Field 'browser' doesn't contain a valid alias configuration
>               /home/mastodon/live/app/javascript/fuzzysort.sass doesn't exist
>             .scss
>               Field 'browser' doesn't contain a valid alias configuration
>               /home/mastodon/live/app/javascript/fuzzysort.scss doesn't exist
>             .css
>               Field 'browser' doesn't contain a valid alias configuration
>               /home/mastodon/live/app/javascript/fuzzysort.css doesn't exist
>             .module.sass
>               Field 'browser' doesn't contain a valid alias configuration
>               /home/mastodon/live/app/javascript/fuzzysort.module.sass doesn't exist
>             .module.scss
>               Field 'browser' doesn't contain a valid alias configuration
>               /home/mastodon/live/app/javascript/fuzzysort.module.scss doesn't exist
>             .module.css
>               Field 'browser' doesn't contain a valid alias configuration
>               /home/mastodon/live/app/javascript/fuzzysort.module.css doesn't exist
>             .png
>               Field 'browser' doesn't contain a valid alias configuration
>               /home/mastodon/live/app/javascript/fuzzysort.png doesn't exist
>             .svg
>               Field 'browser' doesn't contain a valid alias configuration
>               /home/mastodon/live/app/javascript/fuzzysort.svg doesn't exist
>             .gif
>               Field 'browser' doesn't contain a valid alias configuration
>               /home/mastodon/live/app/javascript/fuzzysort.gif doesn't exist
>             .jpeg
>               Field 'browser' doesn't contain a valid alias configuration
>               /home/mastodon/live/app/javascript/fuzzysort.jpeg doesn't exist
>             .jpg
>               Field 'browser' doesn't contain a valid alias configuration
>               /home/mastodon/live/app/javascript/fuzzysort.jpg doesn't exist
>             as directory
>               /home/mastodon/live/app/javascript/fuzzysort doesn't exist
>       /home/mastodon/live/app/javascript/mastodon/features/compose/components/node_modules doesn't exist or is not a directory
>       /home/mastodon/live/app/javascript/mastodon/features/compose/node_modules doesn't exist or is not a directory
>       /home/mastodon/live/app/javascript/mastodon/features/node_modules doesn't exist or is not a directory
>       /home/mastodon/live/app/javascript/mastodon/node_modules doesn't exist or is not a directory
>       /home/mastodon/live/app/javascript/node_modules doesn't exist or is not a directory
>       /home/mastodon/live/app/node_modules doesn't exist or is not a directory
>       /home/mastodon/node_modules doesn't exist or is not a directory
>       /home/node_modules doesn't exist or is not a directory
>       /node_modules doesn't exist or is not a directory
>       looking for modules in /home/mastodon/live/node_modules
>         using description file: /home/mastodon/live/package.json (relative path: ./node_modules)
>           Field 'browser' doesn't contain a valid alias configuration
>           using description file: /home/mastodon/live/package.json (relative path: ./node_modules/fuzzysort)
>             no extension
>               Field 'browser' doesn't contain a valid alias configuration
>               /home/mastodon/live/node_modules/fuzzysort doesn't exist
>             .mjs
>               Field 'browser' doesn't contain a valid alias configuration
>               /home/mastodon/live/node_modules/fuzzysort.mjs doesn't exist
>             .js
>               Field 'browser' doesn't contain a valid alias configuration
>               /home/mastodon/live/node_modules/fuzzysort.js doesn't exist
>             .sass
>               Field 'browser' doesn't contain a valid alias configuration
>               /home/mastodon/live/node_modules/fuzzysort.sass doesn't exist
>             .scss
>               Field 'browser' doesn't contain a valid alias configuration
>               /home/mastodon/live/node_modules/fuzzysort.scss doesn't exist
>             .css
>               Field 'browser' doesn't contain a valid alias configuration
>               /home/mastodon/live/node_modules/fuzzysort.css doesn't exist
>             .module.sass
>               Field 'browser' doesn't contain a valid alias configuration
>               /home/mastodon/live/node_modules/fuzzysort.module.sass doesn't exist
>             .module.scss
>               Field 'browser' doesn't contain a valid alias configuration
>               /home/mastodon/live/node_modules/fuzzysort.module.scss doesn't exist
>             .module.css
>               Field 'browser' doesn't contain a valid alias configuration
>               /home/mastodon/live/node_modules/fuzzysort.module.css doesn't exist
>             .png
>               Field 'browser' doesn't contain a valid alias configuration
>               /home/mastodon/live/node_modules/fuzzysort.png doesn't exist
>             .svg
>               Field 'browser' doesn't contain a valid alias configuration
>               /home/mastodon/live/node_modules/fuzzysort.svg doesn't exist
>             .gif
>               Field 'browser' doesn't contain a valid alias configuration
>               /home/mastodon/live/node_modules/fuzzysort.gif doesn't exist
>             .jpeg
>               Field 'browser' doesn't contain a valid alias configuration
>               /home/mastodon/live/node_modules/fuzzysort.jpeg doesn't exist
>             .jpg
>               Field 'browser' doesn't contain a valid alias configuration
>               /home/mastodon/live/node_modules/fuzzysort.jpg doesn't exist
>             as directory
>               /home/mastodon/live/node_modules/fuzzysort doesn't exist

Basically, the update seems to have gone through anyway and everything is working normally as far as I can tell. However, I have not had these messages before and an incompatible module has, I think, nothing lost in the installation. Therefore, I would like to ask you for your help to eliminate the problem.

**Comment by mdranger** [comment]
Did you check your node version? The error message said:
"error eslint-plugin-promise@6.0.0: The engine "node" is incompatible with this module. Expected version "^12.22.0 || ^14.17.0 || >=16.0.0". Got "12.13.0"

It seems you have node 12.13.0, but the installation requires at least 12.22.0

**Comment by DrSkyfaR** [comment]
Hello @mdranger,
Thanks for your answer, so installed is nodejs 16.15.1-deb-1nodesource1 that says the Debian, even if I ask with node -v. But when I run the same query through the Mastodon user, it tells me that I must have version 12.13.0 installed. Which is obviously not the case. 

Unfortunately I don't know enough about nodejs to be able to isolate the error. Could it be that in some file the value was not changed during an update process of Mastodon ? 

![Screenshot_8](https://user-images.githubusercontent.com/26686045/172817943-c1d3e79c-32f2-4719-9f3f-0d7b0df4213e.png)

**Comment by luiz-m-affonso** [comment]
1. Check your node version at package.json
2. Try this -> $ yarn install --ignore-engines
3. If you have problems, you need to upgrade your version of node.
- nvm current node -v  # Checks your current version
- nvm install version  # Example: nvm install 16.0.0

**Comment by DrSkyfaR** [comment]
Hello @luiz-m-affonso
thank you for the tip with nvm I've always worked without nvm, has worked without problems. But with it is much easier, I have now updated the version to 16.0.5 with nvm and now everything runs through again without errors. 

Thank you very much to you. Always nice to see that there are helpful people like you in the community. 

Have a nice Sunday.
